### PR TITLE
H5Easy: Simplifying xtensor implementation

### PR DESCRIPTION
**Pull request by tdegeus**

### CODE REVIEW

File: include/highfive/h5easy_bits/H5Easy_xtensor.hpp [hunk 1]
```diff
@@ -75,6 +67,22 @@ struct load_impl
...
 
+// front-end
+template <class T>
+inline DataSet dump(File& file,
+                    const std::string& path,
+                    T&& data,
+                    DumpMode mode)
+{
+    if (!file.exist(path)) {
+        return detail::xtensor::dump_impl(file, path, data);
```
Comment (ferdonline) [inline include/highfive/h5easy_bits/H5Easy_xtensor.hpp]: @alexsavulescu data is universal ref, why dont we std::forward(data)? This looks dangerous (copies may be made!)

Comment (ferdonline) [reply include/highfive/h5easy_bits/H5Easy_xtensor.hpp]: Actually, what was the prob with const-ref?

Comment (tdegeus) [reply include/highfive/h5easy_bits/H5Easy_xtensor.hpp]: We could indeed use `std::forward`, though for the moment there is not reason to prefer `T&& data` over `const T& data`. So I can change it to the latter if you prefer.

Comment (alexsavulescu) [reply include/highfive/h5easy_bits/H5Easy_xtensor.hpp]: Yes, we should go with const T& data.

Comment (tdegeus) [reply include/highfive/h5easy_bits/H5Easy_xtensor.hpp]: As you wish: https://github.com/BlueBrain/HighFive/pull/299

**Pull request merged**